examples/llm_serving/launch_model_worker.py

```diff
@@ -22,12 +23,47 @@
 @dataclass(order=True)
 class PrioritizedItem:
     priority: int
-    item: Any=field(compare=False)
+    item: Any = field(compare=False)
+
+
+class RequestType(Enum):
+    INTERACTIVE_JOB = 1
+    BACKGROUND_JOB = 2
+
+
+class WeightedQueue:
+    """
+    A simple wrapper around a PriorityQueue to set
+    a weight for the PriorityQueue.
+    This is supposed to be used with a WRR algorithm.
+    """
+    def __init__(self, weight, queue: asyncio.PriorityQueue = None):
+        if queue is not None:
+            self.queue = queue
+        else:
+            self.queue = asyncio.PriorityQueue()
+        self.weight = weight
+
+    def get_priority_queue(self) -> asyncio.PriorityQueue:
+        return self.queue
+
+    def get_weight(self):
+        return self.weight
 
 
 GenerateItem = namedtuple("GenerateItem", ["uid", "return_queue", "data"])
 LogprobsItem = namedtuple("LogprobsItem", ["uid", "return_queue", "data"])
 
+GENERATE_ITEM_PRIORITY = 0
+LOGPROBS_ITEM_PRIORITY = 50
```

Comment:
This priority can be deprecated now. Let us just use a single-level priority queue instead of the current nested two-level priority queue.

i.e.
Put all requests into a single priority queue. Only set the priority according to their source (i.e. interactive or background). Do not set priorities according to their request endpoint (i.e. generate or logprobs)

Reply:
sounds good. I thought that having two or more queues would provide more flexibility for the future but sure, we can have everything in one queue. Will do.

Reply:
That would default to what it is currently on main with the exception of changing the old priorities based on the source, wouldn't it? Unless I understand it wrong, we still need 2 queues - one for each type of source to avoid starvation. In other words, I need to rotate between interactive and background otherwise background (low priority) can starve if new interactive jobs come in. If instead, I pick 1 background for every n interactive (and background itself has priority given the key), then it would avoid starvation. Am I right or am I missing something?

Reply:
You are right. In the old code, `generate_item` always has a higher priority than `logprob_item`. Now we want to remove this design. `generate_item` and `logprob_item` will have the same priority.
We only assign priority according to sources (iterative vs background). In other words,
Old rule:
logprob_item > generate_item
New rule:
interactive logprob_item = interactive generate_item > background logprob_item = background generate_item